src/TextareaAutosize.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * <TextareaAutosize />
+ */
+
+import React from 'react';
+import emptyFunction from 'react/lib/emptyFunction';
+import autobind from 'autobind-decorator';  //eslint-disable-line no-unused-vars
+import calculateNodeHeight from './calculateNodeHeight';
+
+export default class TextareaAutosize extends React.Component {
+
+  static propTypes = {
+    /**
+     * Current textarea value.
+     */
+    value: React.PropTypes.string,
+
+    /**
+     * Callback on value change.
+     */
+    onChange: React.PropTypes.func,
+
+    /**
+     * Try to cache DOM measurements performed by component so that we don't
+     * touch DOM when it's not needed.
+     *
+     * This optimization doesn't work if we dynamically style <textarea />
+     * component.
+     */
+    useCacheForDOMMeasurements: React.PropTypes.bool
+  }
+
+  static defaultProps = {
+    onChange: emptyFunction,
+    useCacheForDOMMeasurements: false
+  }
+
+  constructor(props) {
+    super(props);
+    this.props = props;
+    this.state = {height: null};
+  }
+
+  render() {
+    let {valueLink, onChange, ...props} = this.props;
+    if (typeof valueLink === 'object') {
+      props.value = this.props.valueLink.value;
+    }
+    props.style = {
+      ...props.style,
+      overflow: 'hidden',
+      height: this.state.height
+    };
+    return <textarea {...props} onChange={this._onChange} />;
+  }
+
+  componentDidMount() {
+    // Timeout seems to be required for all media queries to be applied
```

Comment:
Yes, and... it depends on order of stylesheets vs js code not really on if style are inside media query rules or not.

So if we have
```
    <link rel="stylesheet" href="bundle.css" />
    <script src="bundle.js"></script>
```
then we see correct values in componentDidMount, but in case of
```
    <link rel="stylesheet" href="bundle.css" />
    <script src="bundle.js"></script>
```
we need to wait some time before measuring DOM.

I think we can remove `onNext` call in `componentDidMount` and require users to re-render in case any of styles on page which can influence textarea sizing changed.